docs/tutorials/typescript/digital-ocean-deployment/index.mdx

```diff
@@ -0,0 +1,283 @@
+
+```
+ssh -i ~/.ssh/id_rsa_digitalocean
+```
+
+**Install Node.js**
+
+Once you have SSH'd into your Droplet, you can install Node.js.
+
+Follow these tutorials, for Ubuntu 20.04 or 22.04 respectively:
+
+- [How To Install Node.js on Ubuntu 20.04](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-20-04)
+- [How To Install Node.js on Ubuntu 22.04](https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-ubuntu-22-04)
+
+**Install TypeScript and TS Node**
+
+```
+sudo npm install -g typescript ts-node
+```
+
+## Use a custom domain {#custom-domain}
+
+You need a domain, because you need to have a valid SSL certificate for the Temporal Application to communicate with Temporal Cloud.
+
+Register your domain and ensure the domain's A record points to the droplet's IP address.
+
+## Set up Nginx as a reverse proxy {#set-up-nginx}
+
+SSH into your Droplet and set up Nginx as a reverse proxy.
+
+Follow these tutorials, for Ubuntu 20.04 or 22.04 respectively, including setting up the server blocks:
+
+- [Set up Nginx for Ubuntu 20.04](https://www.digitalocean.com/community/tutorials/how-to-install-nginx-on-ubuntu-20-04)
+- [Set up Nginx for Ubuntu 22.04](https://www.digitalocean.com/community/tutorials/how-to-configure-nginx-as-a-reverse-proxy-on-ubuntu-22-04)
+
+## Create a domain certificate {#create-domain-cert}
+
+Use Let's encrypt certbot to create a certificate for your domain.
+This will effectively enable HTTPS for your domain and your Digital Ocean Droplet.
+
+Follow the tutorial for your Ubuntu version:
+
+- [How To Secure Nginx with Let's Encrypt on Ubuntu 20.04](https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-20-04)
+- [How To Secure Nginx with Let's Encrypt on Ubuntu 22.04](https://www.digitalocean.com/community/tutorials/how-to-secure-nginx-with-let-s-encrypt-on-ubuntu-22-04)
+
+## Create a Temporal Cloud certificate {#create-temporal-cloud-cert}
+
+You will need a seperate certificate to authenticate the communication between your Temporal Client, Temporal Workers, and Temporal Cloud.
+
+You can use tcld to generate the certificate.
+
+Follow the [steps to install tcld](https://docs.temporal.io/cloud/tcld#install-tcld).
+
+Then, follow the [steps to generate a Temporal Cloud certificate using tcld](https://docs.temporal.io/cloud/certificates#use-tcld-to-generate-certificates).
+
+## Clone your repo into the droplet {#clone-repo}
+
+Clone the repo on the droplet
+
+Install dependencies
+
+## Create a `.env` file
+
+Add your Temporal Cloud Namespace, address, certificate PEM and KEY to your projects `.env` file.
+
+```plaintext
+TEMPORAL_CLOUD_ADDRESS=""
+TEMPORAL_CLOUD_NAMESPACE=""
+TEMPORAL_CLOUD_PEM=""
+TEMPORAL_CLOUD_PRIVATE_KEY=""
+```
+
+Your Temporal Client and Worker will need those environment variables to connect to Temporal Cloud.
+
+## Ensure you are using Temporal Cloud {#use-temporal-cloud}
```

Comment:
Another scoping comment -- all of a sudden here we're talking about migrating an application to use Temporal Cloud. I'm not sure whether this belongs in a separate reference article (we could easily publish one for each language), linked under Step 4 above, or as a Step 5 here, but I think we need to more consistent in how we're framing this.

Reply:
It is not migrating though.
You can't use anything but Temporal Cloud when deploying to a DO server like this - 
Ensuring there are proper env vars and they and the code is structured to use them is a critical step - I don't think farming it out to another tutorial is the way to go here -

Reply:
I do get where you are coming from here tho - it feels like it could be a prereq - 
For the scope I am pushing for, I want to keep it for now.

The current tutorial for running Workers with TypeScript feels heavy https://learn.temporal.io/getting_started/typescript/run_workers_with_cloud_typescript/

Reply:
I mean, technically you *could* run a self-hosted Temporal on the same droplet as your workers, right? That's what our upcoming self-hosted tutorial is going to look like. Just want to make sure we're drawing that distinction effectively.